### PR TITLE
Cardview/match

### PR DESCRIPTION
**Pull request by dongeronimo**
Showing the player cards and the discard pile.
Also solved a bug - the player order wasn't random,

### CODE REVIEW

**Comment by dongeronimo** [comment]
image

File: src/card_container.cpp [hunk 1]
```diff
@@ -5,9 +5,21 @@
...
 using namespace std;
 using namespace MyUno;
-shared_ptr<Card> MyUno::CardContainer::Top()
+
+MyUno::CardContainer::CardContainer(vector<shared_ptr<Card>> other)
+{
+    for (auto it = other.begin(); it != other.end(); ++it)
+    {
+        cards.push_back(*it);
+    }
```
Comment (rodrigoaadias) [inline src/card_container.cpp]: for readability, you can also use:
`for (auto card : other)
{
    cards.push_back(card);
}`

File: src/card_container.cpp [hunk 2]
```diff
@@ -5,9 +5,21 @@
...
+}
+
+const shared_ptr<Card> MyUno::CardContainer::Top() const
 {
-    return cards[Count() - 1];
+    if (cards.size() > 0)
```
Comment (rodrigoaadias) [inline src/card_container.cpp]: I guess there is a method called `empty()`. You could also make: 
`return cards.empty() ? nullptr : cards[Count() - 1];`

File: src/match_window.cpp [hunk 3]
```diff
@@ -4,12 +4,28 @@
...
 #include <stdlib.h>
+#include <stdexcept>
 #include <stdio.h>
+#include "card_view.h"
+#include "numeric_card_view.h"
 using namespace std;
+using namespace MyUno;
+shared_ptr<CardView> MyUno::MatchWindow::GetCardView(MyUno::Type type)
+{
+    for (auto cardViewIt = cardViewProcessors.begin(); cardViewIt != cardViewProcessors.end(); ++cardViewIt)
```
Comment (rodrigoaadias) [inline src/match_window.cpp]: same as above: `for(auto cardView : cardViewProcessors)`

**Review by rodrigoaadias — COMMENTED** [review]
Ainda acho que os cards numéricos não deveriam ter um `enum` pra número, poderia ser `int` direto. Acho que isso viola o Open-Closed principle.

**Pull request merged**